app/models/relation.rb

```diff
@@ -241,6 +147,12 @@ def delay
     self[:delay]
   end
 
+  TYPES.each_key do |type|
```

Comment:
I wouldn't want to place it in a scope file. Those files are designed to contain exactly one scope per file, so something to fetch data with. The TYPE constants on the other hand are describing the relation model. It might make sense to place the constants in a separate concern but even then I would keep them separate from the scope.